xrdp/xrdp_bitmap.c

```diff
@@ -160,26 +166,27 @@ xrdp_bitmap_create_with_data(int width, int height,
     self->height = height;
     self->bpp = bpp;
     self->wm = wm;
+    Bpp = 4;
+    switch (bpp)
+    {
+        case 8:
+            Bpp = 1;
+            break;
+        case 15:
+            Bpp = 2;
+            break;
+        case 16:
+            Bpp = 2;
+            break;
```

Comment:
We need two inline functions - sparse_Bpp (returning 1, 2, 3, 4) and aligned_Bpp (returning 1, 2, 4). Repeating the same code is not useful.